[DataGrid] Add `resetPageOnSortFilter` prop that resets the page after sorting/filtering (#16450)
mui/mui-x: docs/data/data-grid/sorting/sorting.md

{{"demo": "ReadOnlySortingGrid.js", "bg": "inline", "defaultCodeOpen": false}}

## Reset pagination

> By default sorting does not change the current page. If you want to get the user back to the first page each time new sort is applied, use `resetPageOnSortFilter` prop.

Reviewer: ```suggestion
By default, sorting does not change the current page. 
To send the user back to the first page when a new sort is applied, use the `resetPageOnSortFilter` prop.
```

{{"demo": "ServerSortingGrid.js", "bg": "inline"}}

:::info

> Combine server-side sorting with [Server-side filtering](/x/react-data-grid/filtering/server-side/) and [Server-side pagination](/x/react-data-grid/pagination/#server-side-pagination) to avoid fetching more data than needed, since you already process the data outside of the Data Grid.

Reviewer: same comments here as above: success callout, small copyedits to the text
